tests/test-module-for-component-test.js

```diff
@@ -298,6 +298,30 @@ test('it allows missing callbacks', function() {
   ok(true, 'no errors are thrown');
 });
 
+let deprecations = [];
+QUnit.module('moduleForComponent: will not raise deprecation if needs is specified', {
+  beforeEach() {
+    originalDeprecate = Ember.deprecate;
+    Ember.deprecate = function() {
+      deprecations.push(arguments);
+    }
+  },
+  afterEach() {
+    Ember.deprecate = originalDeprecate;
+    deprecations = [];
+  }
+});
+
+test('deprecation is not raised', function() {
+  let deprecationId = 'ember-test-helpers.test-module-for-component.test-type';
+  setupRegistry();
+  testModule = new TestModuleForComponent('pretty-color', { needs: ['x:foo'] });
+  notOk(deprecations.find(function(deprecation) {
+    return deprecation.length === 3 && deprecation[2].id === deprecationId;
```

Comment:
would it make sense to only collect the `id` properties in the mocked `deprecate()` function above to simplify the logic here?